operators/pkg/forge/cloudinit.go

```diff
@@ -58,36 +57,52 @@ type writefile struct {
 }
 
 // CloudInitUserData forges the yaml manifest representing the cloud-init userdata configuration.
-func CloudInitUserData(nextcloudBaseURL, webdavUsername, webdavPassword string, publicKeys []string) ([]byte, error) {
-	config := userdata{
-		Users: []user{{
-			Name:       "crownlabs",
-			LockPasswd: false,
-			// The hash of the password ("crownlabs").
-			// You can generate this hash via: "mkpasswd --method=SHA-512 --rounds=4096".
-			Passwd:            "$6$rounds=4096$tBS1sNBpnw6feehB$lS9b7VKH6WMAFOB0SrHCgjD2BKs9CegDe51EiMRWbxQeCVnoGL4u0jNaRsYhvVoBFaRlXZkNsxfFhXvCBaNeQ.",
-			Sudo:              "ALL=(ALL) NOPASSWD:ALL",
+func CloudInitUserData(serviceName string, servicePath string, publicKeys []string) ([]byte, error) {
+	var config userdata
+	if len(serviceName) != 0 && len(servicePath) != 0 {
+		config = userdata{
+			Users: []user{{
+				Name:       "crownlabs",
+				LockPasswd: false,
+				// The hash of the password ("crownlabs").
+				// You can generate this hash via: "mkpasswd --method=SHA-512 --rounds=4096".
+				Passwd:            "$6$rounds=4096$tBS1sNBpnw6feehB$lS9b7VKH6WMAFOB0SrHCgjD2BKs9CegDe51EiMRWbxQeCVnoGL4u0jNaRsYhvVoBFaRlXZkNsxfFhXvCBaNeQ.",
+				Sudo:              "ALL=(ALL) NOPASSWD:ALL",
+				SSHAuthorizedKeys: publicKeys,
+				Shell:             "/bin/bash",
+			}},
+			Network: network{
+				Version: 2,
+				ID0:     interf{DHCP4: true},
+			},
+			Mounts: [][]string{{
+				fmt.Sprintf("%s:%s", serviceName, servicePath),
+				"/media/MyDrive",
+				"nfs",
+				"rw,tcp,hard,intr,rsize=8192,wsize=8192,timeo=14",
+				"0",
+				"0",
+			}},
 			SSHAuthorizedKeys: publicKeys,
-			Shell:             "/bin/bash",
-		}},
-		Network: network{
-			Version: 2,
-			ID0:     interf{DHCP4: true},
-		},
-		Mounts: [][]string{{
-			fmt.Sprintf("%s/remote.php/dav/files/%s", nextcloudBaseURL, webdavUsername),
-			"/media/MyDrive",
-			"davfs",
-			"_netdev,auto,user,rw,uid=1000,gid=1000",
-			"0",
-			"0",
-		}},
-		WriteFiles: []writefile{{
-			Content:     fmt.Sprintf("/media/MyDrive %s %s", webdavUsername, webdavPassword),
-			Path:        "/etc/davfs2/secrets",
-			Permissions: "0600",
-		}},
-		SSHAuthorizedKeys: publicKeys,
+		}
+	} else {
+		config = userdata{
+			Users: []user{{
+				Name:       "crownlabs",
+				LockPasswd: false,
+				// The hash of the password ("crownlabs").
+				// You can generate this hash via: "mkpasswd --method=SHA-512 --rounds=4096".
+				Passwd:            "$6$rounds=4096$tBS1sNBpnw6feehB$lS9b7VKH6WMAFOB0SrHCgjD2BKs9CegDe51EiMRWbxQeCVnoGL4u0jNaRsYhvVoBFaRlXZkNsxfFhXvCBaNeQ.",
+				Sudo:              "ALL=(ALL) NOPASSWD:ALL",
+				SSHAuthorizedKeys: publicKeys,
+				Shell:             "/bin/bash",
+			}},
+			Network: network{
+				Version: 2,
+				ID0:     interf{DHCP4: true},
+			},
+			SSHAuthorizedKeys: publicKeys,
+		}
```

Comment:
You're duplicating the code heavily. You can create an initial, common userdata, then, 
```go
if serviceName != "" && servicePath != "" {
   config.Mounts = ...
}

Reply:
Removed duplicated code